share/eds/eds.go

```diff
@@ -273,3 +277,59 @@ func ReadEDS(ctx context.Context, r io.Reader, root share.DataHash) (eds *rsmt2d
 	}
 	return eds, nil
 }
+
+// ConstructFn aliases a function that creates an ExtendedDataSquare.
+type ConstructFn func(
+	data [][]byte,
+	appVersion uint64,
+	squareSizeBounds int,
+	options ...nmt.Option,
+) (*rsmt2d.ExtendedDataSquare, error)
+
+// MakeExtendedDataSquare extends the given block data, returning the resulting
+// ExtendedDataSquare (EDS). If the data is empty,
+// nil is returned in place of the eds.
+func MakeExtendedDataSquare(
+	data [][]byte,
+	appVersion uint64,
+	squareSizeBounds int,
+	options ...nmt.Option,
+) (*rsmt2d.ExtendedDataSquare, error) {
+	if len(data) == 0 {
+		return nil, nil
```

Comment:
Returning two nils is an anti-pattern. We did that previously in the private code which was okaish. Now that we extracted this as public API func, we should change that.

Reply:
This case does not happen anymore also, because you always check for `IsEmptyBlock` outside of this func

Reply:
Currently yes, but app code relies on the version. What if future version will not check data length. It seems that this is impossible but imo this is a good pattern to check length of the slice which you are going to use. 

Regarding double nil, I agree. Let's then creat an empty Eds?

Reply:
let's please use the app function to check if the block data really is empty and then can return `EmptyRoot` ? 

```
	if app.IsEmptyBlock(data, appVersion) {
		return share.EmptyExtendedDataSquare(), nil
	}

	dataSquare, err := square.Construct(data.Txs.ToSliceOfBytes(), appVersion, squareSizeBounds)
	if err != nil {
		return nil, err
	}

```

That would require changing func sig to take tendermint `types.Data`

Might be import cycle though so see if we can move empty EDS def to eds pkg?